src/Core/Storage/Type/ExternalResource.php

```diff
@@ -57,6 +57,7 @@ public function get(string $reference): string
 		try {
 			$fetchResult = HTTPSignature::fetchRaw($data->url, $data->uid, [HttpClientOptions::ACCEPT_CONTENT => [HttpClientAccept::IMAGE]]);
 		} catch (Exception $exception) {
+			Logger::notice('URL is invalid', ['url' => $data->url, 'error' => $exception]);
```

Comment:
Don't use direct Dependencies per static in these dynamic classes.

Please use instead
```suggestion
			$this->logger->notice('URL is invalid', ['url' => $data->url, 'exception' => $exception]);
```
(I always try to use `exception` as a key for all exceptions, maybe it will help aggregating exceptions later)

and add this as constructor at the start of the class:
```php
	/** @var LoggerInterface */
	protected $logger;

	public function __construct(LoggerInterface $logger)
	{
		$this->logger = $logger;
	}
```

And remove the whole `Logger` dependency of this class

And here's the call:
https://github.com/friendica/friendica/blob/1874a32728142f2c12724562eb122eb1cd1370fe/src/Core/Storage/Repository/StorageManager.php#L211

which you need to change to
```php
$this->backendInstances[$name] = new Type\ExternalResource($this->logger);
```